.github/workflows/build_portable_linux_pytorch_wheels.yml

```
@@ -159,11 +178,55 @@ jobs:
   test_pytorch_wheels:
     if: ${{ needs.generate_target_to_run.outputs.test_runs_on != '' }}
     needs: [build_pytorch_wheels, generate_target_to_run]
-
     uses: ./.github/workflows/test_pytorch_wheels.yml
     with:
       amdgpu_family: ${{ inputs.amdgpu_family }}
       test_runs_on: ${{ needs.generate_target_to_run.outputs.test_runs_on }}
-      cloudfront_url: ${{ inputs.cloudfront_url }}
+      cloudfront_url: ${{ inputs.cloudfront_staging_url }}
       python_version: ${{ inputs.python_version }}
       torch_version: ${{ needs.build_pytorch_wheels.outputs.torch_version }}
+
+  upload_pytorch_wheels:
+    name: Upload PyTorch Wheels to S3
+    needs: [build_pytorch_wheels, generate_target_to_run, test_pytorch_wheels]
+    if: success() && github.repository_owner == 'ROCm'
+    runs-on: ubuntu-24.04
+    env:
+      PACKAGE_DIST_DIR: ${{ github.workspace }}/output/packages/dist
+      S3_BUCKET_PY: "therock-${{ inputs.release_type }}-python"
+      CP_VERSION: ${{ needs.build_pytorch_wheels.outputs.cp_version }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Configure AWS Credentials
+        if: always()
+        uses: aws-actions/configure-aws-credentials@b47578312673ae6fa5b5096b330d9fbac3d116df # v4.2.1
+        with:
+          aws-region: us-east-2
+          role-to-assume: arn:aws:iam::692859939525:role/therock-${{ inputs.release_type }}-releases
+
+      - name: Copy only tested PyTorch wheels from staging to release S3
+        run: |
+          torch_version="${{ needs.build_pytorch_wheels.outputs.torch_version }}"
+          cp_version="${CP_VERSION}"
+          echo "Torch version: $torch_version"
+          # Extract the ROCm suffix from the torch version (everything after '+')
+          rocm_suffix=$(echo "$torch_version" | cut -d '+' -f 2)
+          echo "ROCm suffix: $rocm_suffix"
+
+          for package in torch torchaudio torchvision; do
+            echo "Copying matching files for $package with +$rocm_suffix..."
+            aws s3 cp \
+              s3://${S3_BUCKET_PY}/${{ inputs.s3_staging_subdir }}/${{ inputs.amdgpu_family }}/ \
+              s3://${S3_BUCKET_PY}/${{ inputs.s3_subdir }}/${{ inputs.amdgpu_family }}/ \
+              --recursive \
+              --exclude "*" \
+              --include "${package}-*+${rocm_suffix}-${cp_version}*.whl"
```

```suggestion
              --include "${package}-*+${rocm_suffix}-${cp_version}-linux_x86_64.whl"
```
As you otherwise might copy over Windows builds once they get uploaded to staging.